### DISCUSSION TITLE
Run `brew bump-formula-pr` with GitHub App Token

### DISCUSSION THREAD

**Discussion by 5ouma** [Homebrew/discussions#5129] [Tap maintenance and Homebrew development] [answered]
### Description of issue

I use `brew bump-formula-pr` to keep formulae up-to-date via [dawidd6/action-homebrew-bump-formula](https://github.com/dawidd6/action-homebrew-bump-formula).
I used PAT with `public_repo` permission, and it worked. I wanted to use a GitHub App Token to change the PR opener to a bot. However, it doesn't work with this error, even when I added any permissions.

```
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils/github/api.rb:349:in `raise_error': GitHub API Error: Resource not accessible by integration (GitHub::API::AuthenticationFailedError)
HOMEBREW_GITHUB_API_TOKEN may be invalid or expired; check:
  \e[4mhttps://github.com/settings/tokens\e[24m
	from /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils/github/api.rb:272:in `open_rest'
	from /home/runner/work/_actions/dawidd6/action-homebrew-bump-formula/v3/main.rb:[59](https://github.com/5ouma/homebrew-formula/actions/runs/7895707445/job/21548603535#step:7:63):in `<module:Homebrew>'
	from /home/runner/work/_actions/dawidd6/action-homebrew-bump-formula/v3/main.rb:17:in `<main>'
```

Do you have any ideas to solve this, or is it impossible to use a GitHub App Token?

Related: #4514

**Comment by devantler** [comment]
I need this too. I have also tested it not to work. It is best practice not to use the Personal Access Tokens, as they are getting deprecated in favor of fine-grained tokens and GitHub Apps.

EDIT: Poor choice of wording here. Classic tokens are not deprecated, but GitHub is pushing towards using fine-grained tokens over classic tokens.

**Reply by gromgit** [reply]
> It is best practice not to use the Personal Access Tokens, as they are getting deprecated in favor of fine-grained tokens and GitHub Apps.

I must've missed the announcement. Where/when did GitHub announce this?

**Reply by devantler** [reply]
I mean it is not said directly, but when I read this I believe it is hinted at. Possible as soon as fine-grained tokens have feature parity with classic tokens:

https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#types-of-personal-access-tokens

I updated the initial comment to reflect my poor choice of wording. Sorry about any confusion that might have caused!

**Comment by gromgit** [comment]
FYI, this is an outstanding issue at https://github.com/dawidd6/action-homebrew-bump-formula/issues/66, so anyone who has any concrete ideas should probably help close that issue, and thereby direct benefit all users of the above mentioned action.

**Comment by 5ouma** [comment, marked as answer]
`brew bump --no-fork` allows the use of GitHub App Token.
